torch/onnx/symbolic_opset9.py

```diff
@@ -733,23 +733,46 @@ def replication_pad(g, input, padding):
 replication_pad3d = replication_pad
 
 
-def _interpolate(name, dim, interpolate_mode):
-    def symbolic_fn(g, input, output_size, align_corners=None):
-        sym_help._interpolate_warning(interpolate_mode)
-        align_corners = sym_help._maybe_get_scalar(align_corners)
-        if align_corners:
-            return _unimplemented(name, "align_corners == True")
+def _interpolate(name, dim, interpolate_mode, g, input, output_size, scales, align_corners=None):
+    sym_help._interpolate_warning(interpolate_mode)
+    align_corners = sym_help._maybe_get_scalar(align_corners)
+    if align_corners:
+        return _unimplemented(name, "align_corners == True")
+    if scales is None:
         scales = sym_help._interpolate_size_to_scales(g, input, output_size, dim)
-        return g.op("Upsample", input, scales, mode_s=interpolate_mode)
+    return g.op("Upsample", input, scales, mode_s=interpolate_mode)
+
+
+def _interpolate1d(name, dim, interpolate_mode):
+    def symbolic_fn(g, input, output_size, *args):
+        scales_1, align_corners = sym_help.get_interpolate_attributes(interpolate_mode, args)
+        scales = sym_help._interpolate_get_scales_if_available(g, [scales_1])
```

Comment:
With the above comment update on `get_interpolate_attributes`. It seems the duplication of explicitly writing methods for 1d, 2d, 3d could be removed by 
```
        scales, align_corners = sym_help.get_interpolate_attributes(interpolate_mode, args)
        scales = sym_help._interpolate_get_scales_if_available(g, scales)
```